src/invenio-records-js/controllers/InvenioRecordsCtrl.js

```diff
@@ -235,19 +240,13 @@ function InvenioRecordsCtrl($scope, $rootScope, $q, $window, $location,
       }
 
       // Trigger successful event for action
-      $rootScope.$broadcast('invenio.records.action.success', type);
+      $rootScope.$broadcast('invenio.records.action.success', _actions);
```

Comment:
Hmm ``invenioRecordsActionSuccess`` is only called when all the actions are completed https://github.com/inveniosoftware/invenio-records-js/pull/25/files#diff-a524af0884766736ec0c3785dfcc84d2R305 not for each one. At the end you get the response of the last request and the array of actions ran.